src/diamond/handler/netuitive_handler.py

```diff
@@ -277,32 +277,36 @@ def _add_aws_meta(self):
             resp = urllib2.urlopen(request, timeout=1).read()
             j = json.loads(resp)
```

Comment:
This outside the if statement means you always load the endpoint, but don't always do anything with it. Probably don't want to fetch it if you don't have to.

Reply:
Made a change to only request the endpoint when it's necessary.